rust/lance-index/src/vector/sq/storage.rs

```diff
@@ -188,17 +266,17 @@ impl QuantizerStorage for ScalarQuantizationStorage {
     async fn load_partition(
         reader: &FileReader,
         range: std::ops::Range<usize>,
-        metric_type: MetricType,
+        distance_type: DistanceType,
         metadata: &Self::Metadata,
     ) -> Result<Self> {
         let schema = reader.schema();
         let batch = reader.read_range(range, schema).await?;
 
-        Self::new(
+        Self::try_new(
             metadata.num_bits,
-            metric_type,
+            distance_type,
             metadata.bounds.clone(),
-            batch,
+            [batch].into_iter(),
```

Comment:
this breaks how the chunks were organized, all chunks are merged here

Reply:
this is not merged, it is just make it iterator to be used with `try_new()`.